webapp/src/snippetBuilder.tsx

```diff
@@ -0,0 +1,339 @@
+    }
+
+    backPage() {
+        this.changePage(-1);
+    }
+
+    textInputOnChange = (answerKey: string) => (v: string) => {
+        const answers = this.state.answers;
+        answers[answerKey] = v;
+
+        this.setState({ answers })
+    }
+
+    renderCore() {
+        const { visible, output, projectView, answers, currentQuestion, config } = this.state;
+
+        const actions: sui.ModalButton[] = [
+            {
+                label: lf("Back"),
+                onclick: this.backPage,
+                icon: 'arrow left',
+                className: 'arrow left',
+            },
+            {
+                label: lf("Next"),
+                onclick: this.nextPage,
+                icon: 'arrow right',
+                className: 'arrow right',
+            },
+            {
+                label: lf("Cancel"),
+                onclick: this.hide,
+                icon: "cancel",
+                className: "cancel lightgrey"
+            },
+            {
+                label: lf("Done"),
+                onclick: this.confirm,
+                icon: "check",
+                className: "approve positive"
+            }
+        ];
+
+        const currQ = config.questions[currentQuestion];
+
+        // TODO: Workspace component
+
+        return (
+            <sui.Modal isOpen={visible} className="snippetBuilder" size="large"
+                closeOnEscape={false} closeIcon={false} closeOnDimmerClick={false} closeOnDocumentClick={false}
+                dimmer={true} buttons={actions} header={config.name}
+            >
+                <div>
+                    <div className="list">
+                        {currQ &&
+                            <div>
+                                <div>{currQ.title}</div>
+                                <div className='list horizontal'>
+                                    {currQ.inputs.map((input: any) =>
+                                        <div>
+                                            <sui.Input
+                                                label={input.label && input.label}
+                                                value={answers[input.answerToken] || ''}
+                                                onChange={this.textInputOnChange(input.answerToken)}
+                                            />
+                                        </div>
+                                    )}
+                                </div>
+                            </div>
+                        }
+                    </div>
+                    <div id="snippetBuilderOutput">
+                        {projectView && <md.MarkedContent markdown={this.generateOutputMarkdown(output)} parent={projectView} />}
+                    </div>
+                </div>
+            </sui.Modal>
+        )
+    }
+}
+
+// This will be passed down as a prop
+const staticConfig: any = {
+    name: "Sprite Builder",
+    outputType: 'blocks',
+    output: `enum SpriteKind {
```

Comment:
maybe this should be called "initialOutput" ? Because the final output will be a combination of this plus output from each question

Reply:
Yeah I think that's more clear I updated the format.